x-pack/plugins/observability_solution/apm/public/hooks/use_theme.tsx

```diff
@@ -5,11 +5,74 @@
  * 2.0.
  */
 
-import { useContext } from 'react';
-import { ThemeContext } from 'styled-components';
-import { EuiTheme } from '@kbn/kibana-react-plugin/common';
+import {
+  euiPaletteColorBlind,
+  useEuiTheme,
+  euiPaletteColorBlindBehindText,
+  euiBackgroundColor,
+} from '@elastic/eui';
+import type { EuiThemeComputed } from '@elastic/eui';
+import { COLOR_MODES_STANDARD } from '@elastic/eui';
 
-export function useTheme(): EuiTheme {
-  const theme = useContext(ThemeContext);
-  return theme;
+interface EuiColorsVis {
+  euiColorVis0: string;
+  euiColorVis1: string;
+  euiColorVis2: string;
+  euiColorVis3: string;
+  euiColorVis4: string;
+  euiColorVis5: string;
+  euiColorVis6: string;
+  euiColorVis7: string;
+  euiColorVis8: string;
+  euiColorVis9: string;
+}
+interface EuiColorsVisBehindText {
+  euiColorVis0BehindText: string;
+  euiColorVis1BehindText: string;
+  euiColorVis2BehindText: string;
+  euiColorVis3BehindText: string;
+  euiColorVis4BehindText: string;
+  euiColorVis5BehindText: string;
+  euiColorVis6BehindText: string;
+  euiColorVis7BehindText: string;
+  euiColorVis8BehindText: string;
+  euiColorVis9BehindText: string;
+}
+
+export type UseEuiThemeWithColorsVis = EuiThemeComputed & {
+  euiPaletteColorBlind: EuiColorsVis;
+  euiPaletteColorBlindBehindText: EuiColorsVisBehindText;
+  isDarkColorMode: boolean;
+  euiBackgroundColor: string;
+};
+
+export function useTheme(): UseEuiThemeWithColorsVis {
+  const { euiTheme, colorMode, modifications } = useEuiTheme();
+  const euiPaletteColors = euiPaletteColorBlind();
+  const euiPaletteColorsBlindBehindText = euiPaletteColorBlindBehindText();
+  const isDarkColorMode = colorMode === COLOR_MODES_STANDARD.dark;
+  const uiColorsVis = {
+    euiPaletteColorBlind: euiPaletteColors.reduce((colors, color, index) => {
+      (colors as any)[`euiColorVis${index}`] = color;
+      return colors;
+    }, {} as EuiColorsVis),
+  };
+  const euiColorsVisBehindText = {
+    euiPaletteColorBlindBehindText: euiPaletteColorsBlindBehindText.reduce(
+      (colors, color, index) => {
+        (colors as any)[`euiColorVis${index}BehindText`] = color;
+        return colors;
+      },
+      {} as EuiColorsVisBehindText
+    ),
+  };
+  return {
+    ...euiTheme,
```

Comment:
@fkanout Was there any challenge in returning the same data shape as before? (i.e., `EuiTheme` from '@kbn/kibana-react-plugin/common')

I am wondering, for fixing types, whether it would be better to find a way and make returning `EuiTheme` work or change the places that uses useTheme to follow the new structure.

Reply:
@maryam-saeidi, good question! When I started working on the refactoring, I thought I had only to replace the imports from `styled-components` and use the equivalents from `emotion`. After asking the eui-team, it appeared that it was more than just that.
As you can see from the updates in the [hook](https://github.com/elastic/kibana/pull/186970/files#diff-33a908dfe99292f03aacc6e0e7f5a7a2932042676df29533ec4cb2c5a251f8acR8) the data are coming from different objects with different structure and types. e.g., there is no `euiColorVis` anymore in `emotion` (only an array of values called `euiPaletteColorBlind`)  that I [mapped back to old names](https://github.com/elastic/kibana/pull/186970/files#diff-33a908dfe99292f03aacc6e0e7f5a7a2932042676df29533ec4cb2c5a251f8acR17). Also some value changed regarding the duration (as I commented initially). 

So, to sum up, I'm using the same name of the last element, e.g. `euiColorVis5`, `euiColorVis9BehindText`, to confirm this is the replacement of the old one. But the object, the structure, and the object path to reach that is different. With that _hybrid_ solution, I think we get a good balance as we still know what we are using by keeping the old names, and at the same time, we know it's the new structure from `emotion`